frontend/app-development/features/dataModelling/SchemaEditorWithToolbar/TopToolbar/GenerateModelsButton.tsx

```diff
@@ -8,18 +8,32 @@ import { toast } from 'react-toastify';
 
 export interface GenerateModelsButtonProps {
   modelPath: string;
+  onSetSchemaGenerationErrorMessages: (errorMessages: string[]) => void;
 }
 
-export const GenerateModelsButton = ({ modelPath }: GenerateModelsButtonProps) => {
+export const GenerateModelsButton = ({
+  modelPath,
+  onSetSchemaGenerationErrorMessages,
+}: GenerateModelsButtonProps) => {
   const { data } = useSchemaQuery(modelPath);
-  const { mutate, isPending } = useGenerateModelsMutation(modelPath);
+  const { mutate, isPending } = useGenerateModelsMutation(modelPath, {
+    hideDefaultError: (error) => error?.response?.data?.customErrorMessages ?? false,
+  });
   const { t } = useTranslation();
 
   const handleGenerateButtonClick = () => {
     mutate(data, {
       onSuccess: () => {
         toast.success(t('schema_editor.datamodel_generation_success_message'));
       },
```

Comment:
What happens if we got errors and then correct them, will the old "Alert" still be displayed, since we are not calling `onSetSchemaGenerationErrorMessages([])`? 🤔 😄